src/main/java/org/springframework/data/elasticsearch/core/SearchHitsIterator.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.elasticsearch.core;
+
+import org.springframework.data.util.CloseableIterator;
+
+/**
+ * A {@link SearchHitsIterator} encapsulates {@link SearchHit} results that can be wrapped in a Java 8
+ * {@link java.util.stream.Stream}.
+ *
+ * @author Sascha Woo
+ * @param <T>
+ * @since 4.0
+ */
+public interface SearchHitsIterator<T> extends CloseableIterator<SearchHit<T>> {
+
+	/**
+	 * @return the number of total hits.
+	 */
+	long getTotalHits();
+}
```

Comment:
what about other information like the total hits relation or the max score (check what's in the `SearchHits` class)?

Reply:
The scroll methods just return `ScrolledPage` at the moment.

Reply:
wouldn't it make sense to rework the `ScrolledPage` stuff then as well?  
Looking at the rest client (https://github.com/spring-projects/spring-data-elasticsearch/blob/master/src/main/java/org/springframework/data/elasticsearch/core/ElasticsearchRestTemplate.java#L260-L269) the `SearchDocumentResponse` passed into the converter has all the information.

In the mapper (https://github.com/spring-projects/spring-data-elasticsearch/blob/master/src/main/java/org/springframework/data/elasticsearch/core/convert/MappingElasticsearchConverter.java#L141-L168) you see just below the method used to return the deprecated `AggregatedPageImpl` how the `SearchHits`object is built.

So maybe this would be a good chance to clean up the methods here a little bit

Reply:
Sure, will take a look at it later.

Reply:
btw `SearchHits` already has a property for the scrollid, and `SearchPage` wraps a `SearchHits`. So these eventually can be reused for changing the internal implementation of `StreamQueries`